### DISCUSSION TITLE
Optimizing Blog Post Updates with ISR in a Portfolio Website

### DISCUSSION THREAD

**Discussion by AnmolSaini16** [vercel/next.js#65955] [Help]
### Summary

I have created a portfolio website that includes a blog page. To fetch the blog posts, I've integrated the dev.to API and used the Incremental Static Regeneration (ISR) approach for faster loading of blogs, setting the revalidation period to 30 minutes. However, there's an issue: if I publish a new post, users must visit the site twice or refresh the page twice after 30 minutes to see the update (a default behavior in Next.js for server revalidation). 

This doesn't seem very ideal. Any suggestions for a better data-fetching approach? I'm considering using on-demand ISR with an API to manually revalidate the blog posts.

### Additional information

```js
// action.ts
"use server"

export const getPosts = async (): Promise<IPost[] | undefined> => {
  let postsResponse;
  try {
    const headers = new Headers();
    headers.append("api-key", process.env.DEV_TO_API_KEY!);

    postsResponse = await fetch("https://dev.to/api/articles/me/published", {
      headers,
      next: { revalidate: 1800 },
    });
  } catch (error) {
    console.log(error);
  }

  if (postsResponse?.ok) {
    return postsResponse.json();
  } else {
    console.log(
      `HTTP Response Code: ${postsResponse?.status}, Message: ${postsResponse?.statusText}`
    );
  }
};
```

**Comment by hasan-mohammad-bd** [comment]
You can create an API endpoint to trigger revalidation and update the cache when a new post is published. This way, users will see the latest posts without needing to refresh the page twice.

**Reply by AnmolSaini16** [reply]
I had the same implementation in mind but was curious about how others approach this feature. Creating a separate API for such a small website sometimes feels redundant.
